gamemodes/terrortown/entities/weapons/weapon_tttbase.lua

```diff
@@ -79,12 +79,21 @@ end
 -- random weapon by a ttt_random_weapon entity.
 SWEP.AutoSpawnable = false
 
+-- Set to one of the WEAPON_TYPE_ flags to define on which spawn this waeapon should
+-- spawn. The flag AutoSpawnable has to be true to make this weapon spawnable on
+-- the map.
+SWEP.spawnType = nil
+
 -- Set to true if weapon can be manually dropped by players (with Q)
 SWEP.AllowDrop = true
 
 -- Set to true if weapon kills silently (no death scream)
 SWEP.IsSilent = false
 
+-- Set this to a number greater than 0 if you want to autospawn random ammo
+-- in close proximity to this weapon when spawned.
+SWEP.autoAmmoAmount = 0
+
```

Comment:
I see what you mean, but all variables added in the last few years are lowercase.

I'm pretty sure you set the default value in your code. And you give it a value if it is nil. Therefore it should be nil, if not set.

Reply:
well, the first is kinda true, we implemented not_buyable etc. outside of that class.

I checked the second and we currently derive the default via .kind or my given default WEAPON_TYPE_SPECIAL, so i guess that's okay. I know that you would normally do it via code, but I meant, that we can give here an inherited default, in case they dont update their weapons.

You can resolve this, if the current behaviour is what you expeckt 😄